src/Ad5001/BetterGen/populator/DeadbushPopulator.php

```diff
@@ -35,26 +38,31 @@ public function populate(ChunkManager $level, $chunkX, $chunkZ, Random $random)
 		for($i = 0; $i < $amount; $i++) {
 			$x = $random->nextRange($chunkX * 16, $chunkX * 16 + 15);
 			$z = $random->nextRange($chunkZ * 16, $chunkZ * 16 + 15);
+			if(!in_array($level->getChunk($chunkX, $chunkZ)->getBiomeId(abs($x % 16), ($z % 16)), [40, 39, Biome::DESERT])) continue;
 			$y = $this->getHighestWorkableBlock($x, $z);
-			if ($y !== -1 && $level->getBlockIdAt($x, $y - 1, $z ) == Block::SAND) {
-				$level->setBlockIdAt($x, $y + 1, $z, Block::DEAD_BUSH);
-				$level->setBlockDataAt($x, $y + 1, $z, 1);
+			if ($y !== -1) {
+				$level->setBlockIdAt($x, $y, $z, Block::DEAD_BUSH);
+				$level->setBlockDataAt($x, $y, $z, 1);
 			}
 		}
 	}
-	
-	/*
+
+	/**
 	 * Gets the top block (y) on an x and z axes
-	 * @param $x int
-	 * @param $z int
+	 * @param $x
+	 * @param $z
+	 * @return int
 	 */
-	protected function getHighestWorkableBlock($x, $z) {
-		for($y = 127; $y >= 0; -- $y) {
+	private function getHighestWorkableBlock($x, $z){
+		for($y = Level::Y_MAX - 1; $y > 0; --$y){
 			$b = $this->level->getBlockIdAt($x, $y, $z);
-			if ($b !== Block::AIR and $b !== Block::LEAVES and $b !== Block::LEAVES2) {
+			if($b === Block::DIRT or $b === Block::GRASS or $b === Block::SAND or $b === Block::SANDSTONE or $b === Block::HARDENED_CLAY or $b === Block::STAINED_HARDENED_CLAY){
 				break;
+			}elseif($b !== Block::AIR){
```

Comment:
You don't want to replace transparent blocks ;) (glass, water, pressure plates, doors...)